### PR TITLE
Feature/models

### PR DESCRIPTION
**Pull request by EugeneLagonski**
Added Actor, Director, Movie models with API.

### CODE REVIEW

**Comment by EugeneLagonski** [comment]
RFR

File: main_app/migrations/0003_auto_20181214_0847.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.1.4 on 2018-12-14 08:47
```
Comment (andreybulygin) [inline main_app/migrations/0003_auto_20181214_0847.py]: Let's squash migrations whenever possible. 
E.g. in this case we can afford having 1 migration only I believe. 
TODO: discuss this in voice when sync-ing up.

File: main_app/models.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    name = models.CharField(max_length=80)
+
+    def __str__(self):
+        return self.name
+
+
+class Movie(mixins.Timestamps):
+    title = models.CharField(max_length=80)
+    director = models.ForeignKey(Director, on_delete=models.CASCADE, related_name='movies')
+    actors = models.ManyToManyField(Actor)
```
Comment (andreybulygin) [inline main_app/models.py]: Let's add "actor's role within the movie"

Comment (andreybulygin) [reply main_app/models.py]: also, a sign, whether this role is `primary` in the movie. Several roles can be primary for a movie.

File: main_app/tests.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        self.director3 = models.Director.objects.create(name='Name3')
+        self.director4 = models.Director.objects.create(name='Name4')
+
+    def test_get_valid_single_director(self):
+        response = self.client.get(reverse(
+            'director-detail',
+            kwargs={'pk': self.director3.pk}))
+        director = models.Director.objects.get(pk=self.director3.pk)
+        serializer = serializers.DirectorSerializer(director)
+        self.assertEqual(response.data, serializer.data)
```
Comment (andreybulygin) [inline main_app/tests.py]: :thumbsup: Let's do this `data` check everywhere else

File: main_app/tests.py [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+    def test_get_invalid_single_movie(self):
+        response = self.client.get(reverse(
+            'movie-detail',
+            kwargs={'pk': 0}))
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class MovieUpdateTest(APITestCase):
+    def setUp(self):
+        actor1 = models.Actor.objects.create(name='Actor1')
```
Comment (andreybulygin) [inline main_app/tests.py]: I'm not sure that the current setUp will be useful (reusable) for many tests.
It's often better to prepare certain data for a certain `test_method()`.
Let's keep it for now, but please pay attention.

**Comment by EugeneLagonski** [comment]
RFR

File: main_app/serializers.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            director = models.Director.objects.get(id=director_id)
+            movie = models.Movie.objects.create(title=title, director=director)
+            if 'actors' in self.initial_data:
+                actors = self.initial_data.get('actors')
+                for actor in actors:
+                    actor_id = actor.get('actor_id')
+                    character_name = actor.get('character_name')
+                    primary = actor.get('primary')
+                    actor_instance = models.Actor.objects.get(pk=actor_id)
+                    models.Role(movie=movie, actor=actor_instance, character_name=character_name, primary=primary)
```
Comment (andreybulygin) [inline main_app/serializers.py]: minor: alignment is not nice, use multilines for long expressions, e.g:

```python
models.Role(
  movie=movie,
  ...
  primary=primary
)
```

**Pull request merged**